float64 tx
float64 ty
float64 tz
float64 tR
float64 tP
float64 tY

